### ISSUE TITLE
Structural commands cannot link to entities inside an anonymous namespace

### ISSUE THREAD

**Issue by GustavoQueipo** [doxygen/doxygen#8437]
**Describe the bug**
Entities within C++ anonymous namespaces cannot be documented from somewhere else. Structural commands within documentation blocks located in separate files fail to link to such entities.

Assuming that `EXTRACT_ANON_NSPACES = YES` in the config file,
Given an anonymous namespace with some function inside:
```c++
// file.cpp

namespace {
    void f() {}
}
```
And this special documentation block in a separate file:
```
// another_file.dox

/**
    \fn void anonymous_namespace{file.cpp}::f()
        Description of f().
*/
```
Doxygen correctly extracts the anonymous namespace and the function within. However, the documentation provided separately is ignored. The `{file.cpp}` part of the link does not seem to work.

Accordingly, the output log shows this warning:
> Generating docs for namespace anonymous_namespace{file.cpp}
> (...)/file.cpp:4: warning: Member f() (function) of anonymous_namespace{file.cpp} is not documented.

A similar issue occurs when trying to link to specialized class template member functions.

**Expected behavior**
Structural commands correctly link to entities within anonymous namespaces.

**To Reproduce**
See description of the bug above.

**Version**
Doxygen 1.9.1
Windows 10, 64-bit

**Additional context**
In modern C++, non-member functions only used within a single translation unit are placed inside anonymous namespaces to give them internal linkage, this is, so that the scope of the functions is restricted to the file where they are defined. This is the typical case for (implementation-only) helper functions.

**Comment by albert-github** [comment]
At first glance looks like to be related to the problem solved in the proposed pull request #8427, though it looks like this is not solved by this.

**Comment by albert-github** [comment]
With doxygen version 1.9.1 I get the warning like:
```
file.cpp:4: warning: Member f() (function) of  anonymous_namespace{file.cpp} is not documented.
```
and `f` is not documented.

With the current doxygen version (1.11.0) there is no warning and the function `f` is documented.
(looks like to be fixed in version 1.9.5)

I think the problem has been solved in the mean time.
